### PR TITLE
feat/GRW-1016/CPAS error tracking

### PR DESCRIPTION
**Pull request by marian-buc**
## What?
Added error tracking using the new hook on CheckoutPayment as well as useAdyenCheckout

## Why?
To track checkout errors on Google Analytics besides Datadog
Thoughts:

- Should we have a separate trackErrorEvent() and have the hook call the correct function?
- What data needs to be sent along with each error? Do we need the selectedBundle data like in trackOfferEvent()?
- error optional parameter typedef? 

**Ticket(s): [GRW-1016]**

[GRW-1016]: https://hedvig.atlassian.net/browse/GRW-1016?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/client/utils/tracking/gtm.ts [hunk 1]
```diff
@@ -117,6 +117,20 @@ export enum EventName {
   ContactInformationPageOpen = 'contact_information_page_open',
 }
 
+export enum ErrorEventName {
+  CheckoutErrorPaymentTokenMutation = 'checkout_error_payment_token_mutation',
+  CheckoutErrorPaymentTokenID = 'checkout_error_payment_token_id_missing',
```
Comment (robinandeer) [inline src/client/utils/tracking/gtm.ts:122]:
```suggestion
  CheckoutErrorPaymentTokenIDMissing = 'checkout_error_payment_token_id_missing',
```

File: src/client/utils/tracking/trackOfferEvent.ts [hunk 2]
```diff
@@ -28,6 +28,8 @@ type OptionalParameters = {
   phoneNumberData?: GTMPhoneNumberData
   quoteCartId?: string
   memberId?: string
+  // TODO Better typedef?
+  error?: Error | any
```
Comment (robinandeer) [inline src/client/utils/tracking/trackOfferEvent.ts:32]: why is any needed?

**Review by robinandeer — APPROVED** [review]
I think this looks great - just some small comments!

File: src/client/utils/tracking/trackOfferEvent.ts (hunk 2 above)
Comment (marian-buc) [reply src/client/utils/tracking/trackOfferEvent.ts:32]: well, Error is a generic interface and catch blocks return (error: unknown) objects which are different depending on what's called inside, so I figured it's a good measure for usability. Looking forward to a better solution

Comment (robinandeer) [reply src/client/utils/tracking/trackOfferEvent.ts:32]: I would prefer `Error | unknown` if simply `Error` doesn't work -- to better reflect what this actually us :)

**Review by robinandeer — DISMISSED** [review]
Consider narrowing the type - otherwise looks good!

File: src/client/utils/tracking/trackOfferEvent.ts (hunk 2 above)
Comment (marian-buc) [reply src/client/utils/tracking/trackOfferEvent.ts:32]: cool, makes sense

**Review by dendoesit — APPROVED** [review]
⭐

**Pull request merged**